barnacle/build.py

```diff
@@ -0,0 +1,52 @@
+# barnacle imports
+import barnacle
+import barnacle.helper
+
+# third party imports
+import argparse
+import docker
+import dockerpty
+import os
+import yaml
+
+def _get_dockerfile(dir):
+    '''
+    helper method to find dockerfiles in a directory
+    '''
+    docker_file = 'Dockerfile'
+    files = []
+    for root, dirs, file in os.walk(dir):
+        if docker_file in file:
+            files.append(root)
+    return files
+
+def build(conf, os=None, all=False):
+    '''
+    build docker images
+    '''
+    def _build_image(path, os):
+        for line in client.build(path=path,
+                                 tag='salt' + os,
+                                 stream=True):
+            print(line.split('":"')[1])
```

Comment:
This might blow up if client.build fails. Better error handling might be good here.